spec/controllers/assignment_invitations_controller_spec.rb

```diff
@@ -68,6 +68,30 @@
       end
     end
 
+    context 'github repository with the same name already exists' do
+      let(:assignment_repo) { AssignmentRepo.new(assignment: assignment, user: user) }
+
+      before do
+        assignment_repo.create_github_repository
+      end
+
+      it 'creates a new assignment repo' do
+        patch :accept_invitation, id: invitation.key
+        expect(user.assignment_repos.count).to eql(1)
+      end
+
+      it 'new repository name has expected suffix' do
+        patch :accept_invitation, id: invitation.key
+
+        expect(WebMock).to have_requested(:post, github_url("/organizations/#{organization.github_id}/repos"))
+          .with(body: /^.*#{assignment_repo.repo_name}-1.*$/)
+      end
+
+      after do
+        assignment_repo.destroy_github_repository
+      end
+    end
+
```

Comment:
Thanks @tarebyte, this looks pretty clear.

However, seems that I still need to put `assignment_repo` in the let as `assignment_repo.repo_name` is referred in the second test case:

```
it 'new repository name has expected suffix' do
  expect(WebMock).to have_requested(:post, github_url("/organizations/#{organization.github_id}/repos"))
    .with(body: /^.*#{assignment_repo.repo_name}-1.*$/)
end
```